File: airbrush/src/main/java/com/subgarden/airbrush/Palette.java
package com.subgarden.airbrush;

import android.support.annotation.ColorInt;

/**
 * A color palette describing the color of the four corners of an image, and its center color.
 *
 * @author Fredrik Larsen <f@subgarden.com>
 */
public class Palette {
    public @ColorInt int topLeft;
    public @ColorInt int topRight;
    public @ColorInt int bottomLeft;
    public @ColorInt int bottomRight;

    public Palette(@ColorInt int topLeft,
                   @ColorInt int topRight,
                   @ColorInt int bottomLeft,
                   @ColorInt int bottomRight) {
        this.topLeft = topLeft;
        this.topRight = topRight;
        this.bottomLeft = bottomLeft;
        this.bottomRight = bottomRight;
    }

}

File: airbrush/src/main/java/com/subgarden/airbrush/Palette.kt
package com.subgarden.airbrush

import android.support.annotation.ColorInt

/**
 * A color palette describing the color of the four corners of an image.
 *
 * @author Fredrik Larsen (fredrik@subgarden.com)
 */
data class Palette(@param:ColorInt @field:ColorInt val topLeft: Int,
                   @param:ColorInt @field:ColorInt val topRight: Int,
                   @param:ColorInt @field:ColorInt val bottomRight: Int,
                   @param:ColorInt @field:ColorInt val bottomLeft: Int)
